### ISSUE TITLE
rollup with vue library dependecies

### ISSUE THREAD

**Issue by dsvanidze** [team-innovation/vue-sfc-rollup#45]
Hi @mgdodge,

Thank you for your response in the issue https://github.com/team-innovation/vue-sfc-rollup/issues/44, but could you give me advice, how to extend vue-sfc-rollup, when my sfc/library depends on other Vue libraries like Vuetify, VeeValidate, etc.?

People use a lot of Vue libraries to build their components and applications. It is normal to use `./src/plugins/*` to add own configurations to Vue's third party libraries. They are listed then in `main.ts` to use them globally in the Vue application. How can I use `vue-sfc-rollup` so that I can use a bundled component in other projects, which maybe have or not all these libraries with my custom configurations - colors, font-style, etc.

Here is my main.ts for Vue Application.
```js
// main.ts
import Vue from 'vue'
import App from './App.vue'

import vuetify from '@/plugins/vuetify'
import VeeValidate from '@/plugins/vee-validate'
Vue.use(VeeValidate)

Vue.config.productionTip = false

new Vue({
  vuetify,
  router,
  render: h => h(App),
}).$mount('#app')
```

How and where can I implement these plugin configurations and `Vue.use(...)` when I would like to bundle an SFC with these dependencies and customizations?

**Comment by mgdodge** [comment]
As per my response in #44, the [linked issue](https://github.com/vuetifyjs/vuetify-loader/issues/70#issuecomment-562007860) contains all you need to get vue-sfc-rollup to bundle vuelidate based libraries. Also, your [linked repo](https://github.com/dsvanidze/vue-sfc-rollup/tree/master/examples/sfc-rollup-vuetify) seems to have been updated with rollup-plugin-vuetify, which accomplishes the same thing.

Vue-sfc-rollup as a utility is working exactly as designed, to scaffold a basic setup for writing/distributing sfc components/libraries to npm. However, it is beyond the scope of this utility to add support for specific third party libraries.

Issues logged here should be for bugs, which this is not. This is asking how to fork the repo to add your own functionality. While I applaud the effort, I cannot support every use case, for reasons already stated. As your repo shows, you have made great progress. In fact, you have ended up a step beyond where I did while trying out your fork.

**Comment by Wilson-Lim** [comment]
@dsvanidze , do u manage to make it work on [your repo](https://github.com/dsvanidze/vue-sfc-rollup/tree/master/examples/sfc-rollup-vuetify) ?

**Comment by dsvanidze** [comment]
@Wilson-Lim Yes, I will update my repo on Monday and you will see changes.

**Comment by Wilson-Lim** [comment]
@dsvanidze thanks, i tried ur existing repo, after importing to other project ,project not running at all.

**Comment by dsvanidze** [comment]
@Wilson-Lim I changed several things in my repository [https://github.com/dsvanidze/vue-sfc-rollup/tree/master/examples/sfc-rollup-vuetify](https://github.com/dsvanidze/vue-sfc-rollup/tree/master/examples/sfc-rollup-vuetify) and added documentation to make reproduction easier. I tested it in my other projects and it works.

I did not need multiple components and treeshaking for my real project, therefore I did not try it. However, I am sure it could work well with small changes.
